src/sysupdate/sysupdate.c

```diff
@@ -163,6 +165,16 @@ static int context_read_definitions(
         }
 
         for (size_t i = 0; i < c->n_transfers; i++) {
+                r = strv_push(&c->changelog, c->transfers[i]->changelog);
+                if (r < 0)
+                        return r;
+                r = strv_push(&c->appstream, c->transfers[i]->appstream);
+                if (r < 0)
+                        return r;
+
+                c->changelog = strv_uniq(c->changelog);
+                c->appstream = strv_uniq(c->appstream);
```

Comment:
plese move these two lines outside the loop

Reply:
and probably this entire loop needs to move into the upper loop where the Transfer is being created, no? Is there any reason for this loop to exist separately?